utils/FactoryDeployerHelper.ts

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: 2024 IEXEC BLOCKCHAIN TECH <contact@iex.ec>
+// SPDX-License-Identifier: Apache-2.0
+
+import { ContractFactory } from '@ethersproject/contracts';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { deployments, ethers } from 'hardhat';
+import { GenericFactory, GenericFactory__factory } from '../typechain';
+import { getBaseNameFromContractFactory } from './deploy-tools';
+const { EthersDeployer: Deployer, factoryAddress } = require('../utils/FactoryDeployer');
+
+export class FactoryDeployerHelper {
+    salt: string;
+    init: any;
+    genericFactory: GenericFactory;
+
+    constructor(owner: SignerWithAddress, salt: string) {
+        this.salt = salt;
+        this.init = new Deployer(owner);
+        this.genericFactory = GenericFactory__factory.connect(factoryAddress, owner);
+    }
+
+    /**
+     * Deploy through a GenericFactory a contract [and optionally trigger call]
+     */
+    async deployWithFactory(
+        contractFactory: ContractFactory,
+        constructorArgs?: any[],
+        call?: string,
+    ) {
+        await this.init.ready(); // Deploy GenericFactory if not already done
+        let bytecode = contractFactory.getDeployTransaction(...(constructorArgs ?? [])).data;
+        if (!bytecode) {
+            throw new Error('Failed to prepare bytecode');
+        }
+        let contractAddress = call
+            ? await this.genericFactory.predictAddressWithCall(bytecode, this.salt, call)
+            : await this.genericFactory.predictAddress(bytecode, this.salt);
+        const previouslyDeployed = (await ethers.provider.getCode(contractAddress)) !== '0x';
+        if (!previouslyDeployed) {
+            call
+                ? await this.genericFactory
+                      .createContractAndCall(bytecode, this.salt, call)
+                      .then((tx) => tx.wait())
+                : await this.genericFactory
+                      .createContract(bytecode, this.salt)
+                      .then((tx) => tx.wait());
```

Comment:
```suggestion
            const tx = call
                ? await this.genericFactory.createContractAndCall(bytecode, this.salt, call)
                : await this.genericFactory.createContract(bytecode, this.salt);
            tx.wait();
```

Reply:
Could idea to remove duplication, please check a slightly different (missing `await` in suggestion) [Factorize create & predict contract](https://github.com/iExecBlockchainComputing/PoCo/pull/58/commits/c01bc8f44b714562182cb0967b44e36f7bc123e8) :ear: